tools/seed/dev_seed.py

```
@@ -163,6 +206,27 @@ def wait_for_postgres(py: Path, root: Path, db_url: str, timeout_seconds: int =
+    parsed_host = db_url.split("@", 1)[-1].split("/", 1)[0].split(":", 1)[0]
+    if parsed_host in {"localhost", "127.0.0.1"}:
+        docker_ip = _docker_postgres_ip()
+        if docker_ip:
+            print(
+                f"[dev-seed] localhost did not accept Postgres connections; retrying via container IP {docker_ip} ..."
+            )
+            fallback_url = _replace_database_host(db_url, docker_ip)
+            env["DATABASE_URL"] = to_asyncpg_url(fallback_url)
+            deadline = time.time() + 30
+            while time.time() < deadline:
+                result = subprocess.run([str(py), "-c", checker], cwd=root, env=env)
```

Ruff (0.15.11)

[error] `subprocess` call: check for execution of untrusted input

(S603)